src/lib.rs

```diff
@@ -500,6 +500,64 @@ impl CertificateDer<'_> {
     }
 }
 
+/// A TLS-encoded Encrypted Client Hello (ECH) configuration list (`ECHConfigList`); as specified in
+/// [draft-ietf-tls-esni-18 §4](https://datatracker.ietf.org/doc/html/draft-ietf-tls-esni-18#section-4)
+#[derive(Clone)]
+pub struct EchConfigListBytes<'a>(BytesInner<'a>);
+
+impl EchConfigListBytes<'_> {
+    /// Converts this config into its owned variant, unfreezing borrowed content (if any)
+    #[cfg(feature = "alloc")]
+    pub fn into_owned(self) -> EchConfigListBytes<'static> {
+        EchConfigListBytes(self.0.into_owned())
+    }
+}
+
+impl fmt::Debug for EchConfigListBytes<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        hex(f, self.as_ref())
+    }
+}
+
+impl PartialEq for EchConfigListBytes<'_> {
+    fn eq(&self, other: &Self) -> bool {
+        self.as_ref().eq(other.as_ref())
+    }
+}
```

Comment:
Why can't we derive this? (Same for `Eq`.)

Reply:
There's no `Eq`/`PartialEq` implemented for `BytesInner` (Prev `DerInner`). The pre-existing `Der` type that also wraps `BytesInner` is implementing `Eq`/`PartialEq` by hand as well as a result. 

I think it would be reasonable to implement `PartialEq` for `BytesInner` (unless there's a reason you avoided it when adding `DerInner` I'm missing?) - it's a crate-internal type so I don't think we need to worry about folks comparing a `BytesInner` from a `Der` with a `BytesInner` from a `EchConfigListBytes`. 

I'll tack on a commit changing this up for both `Der` and `EchConfigListBytes`.

Reply:
> I'll tack on a commit changing this up for both Der and EchConfigListBytes.

Done.